File: dominio/modelos orijinales/BackOption.java
package servicebackmenu2;

import servicebackmenu2.ListMenus;

public class BackOption extends servicebackmenu2.MessagesOption {
    
    private boolean executed;
    public String previousMenu;

    public BackOption(String previousMenu) {
        super("Atras");
        this.executed = false;
        this.previousMenu = previousMenu;
    }

    public void interact() {
        this.executed = true;
        ListMenus.getInstance().getMenu(this.previousMenu).interact();
    }

    protected boolean isExecuted() {
        return this.executed;
    }
    
    protected void reset() {
        this.executed = false;
    }
}

File: conect4/src/main/resources/static/js/memus/models/generics/BackOption.js
import Option from "./Option.js";
import ListMenus from "./ListMenus.js";
import Message from "../../../Messages/Message.js";

class BackOption extends Option {
    
    title;
    previousMenu;

    constructor(namePreviousMenu) {
        super();
        this.namePreviousMenu = namePreviousMenu;
        this.previousMenu = ListMenus.getInstance().getMenu(namePreviousMenu);
    }

    updateTitle() {
        this.title = Message.getInstance().getMessages("Common").back;
    }


    execute(){
        return this.previousMenu;
    }

}

export default BackOption;